### PR TITLE
Remove value policy from core

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/1767

No changelog because the policy was Internal only

### CODE REVIEW

File: sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp [hunk 1]
```diff
@@ -427,59 +427,4 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
...
-      std::map<std::string, std::string> QueryValues;
-    };
-
-    /**
-     * @brief Value policy.
-     *
-     * @details Applies key-value pair values to each HTTP request (either HTTP headers or query
-     * parameters).
-     */
-    class ValuePolicy : public HttpPolicy {
```
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp:446]: FYI, this is the type of change we wouldn't (easily) be able to make post-GA. Even if we no longer have any users within the repo, if we GA'd a package that used it at some point, we'd have to keep such internal types around.

**Review by ahsonkhan — APPROVED** [review]
looks good. consider holding off on merging until tomorrow.

**Comment by ahsonkhan** [comment]
https://dev.azure.com/azure-sdk/public/_build/results?buildId=786398&view=logs&j=70e13c24-1688-50ce-17e7-be375e71fa16&t=62d2e991-4d61-5843-f80c-7b32310abe2c&l=3074

Looks similar to https://github.com/Azure/azure-sdk-for-cpp/issues/1889
```text
2021-03-16T16:46:47.3009923Z test 139
2021-03-16T16:46:47.3010691Z         Start 139: azure-core.Test/TransportAdapter.get/libCurl
2021-03-16T16:46:47.3011082Z 
2021-03-16T16:46:47.3012120Z 139: Test command: /mnt/vss/_work/1/s/build/sdk/core/azure-core/test/ut/azure-core-test "--gtest_filter=Test/TransportAdapter.get/libCurl" "--gtest_also_run_disabled_tests"
2021-03-16T16:46:47.3012996Z 139: Test timeout computed to be: 10000000
2021-03-16T16:46:47.3034068Z 139: Note: Google Test filter = Test/TransportAdapter.get/libCurl
2021-03-16T16:46:47.3034829Z 139: [==========] Running 1 test from 1 test suite.
2021-03-16T16:46:47.3035831Z 139: [----------] Global test environment set-up.
2021-03-16T16:46:47.3036711Z 139: [----------] 1 test from Test/TransportAdapter
2021-03-16T16:46:47.3037601Z 139: [ RUN      ] Test/TransportAdapter.get/libCurl
2021-03-16T16:46:47.4658787Z 139: unknown file: Failure
2021-03-16T16:46:47.4660848Z 139: C++ exception with description "Fail to get a new connection for: httpbin.org. Couldn't resolve host name" thrown in the test body.
2021-03-16T16:46:47.4663598Z 139: [  FAILED  ] Test/TransportAdapter.get/libCurl, where GetParam() = 48-byte object <70-96 4F-45 EE-55 00-00 07-00 00-00 00-00 00-00 6C-69 62-43 75-72 6C-00 73-70 6F-72 74-41 64-61 50-83 4F-45 EE-55 00-00 40-83 4F-45 EE-55 00-00> (161 ms)
2021-03-16T16:46:47.4665129Z 139: [----------] 1 test from Test/TransportAdapter (161 ms total)
2021-03-16T16:46:47.4665801Z 139: 
2021-03-16T16:46:47.4666575Z 139: [----------] Global test environment tear-down
2021-03-16T16:46:47.4667274Z 139: [==========] 1 test from 1 test suite ran. (161 ms total)
2021-03-16T16:46:47.4668038Z 139: [  PASSED  ] 0 tests.
2021-03-16T16:46:47.4668706Z 139: [  FAILED  ] 1 test, listed below:
2021-03-16T16:46:47.4670338Z 139: [  FAILED  ] Test/TransportAdapter.get/libCurl, where GetParam() = 48-byte object <70-96 4F-45 EE-55 00-00 07-00 00-00 00-00 00-00 6C-69 62-43 75-72 6C-00 73-70 6F-72 74-41 64-61 50-83 4F-45 EE-55 00-00 40-83 4F-45 EE-55 00-00>
2021-03-16T16:46:47.4671434Z 139: 
2021-03-16T16:46:47.4671987Z 139:  1 FAILED TEST
2021-03-16T16:46:47.4673107Z 139/165 Test #139: azure-core.Test/TransportAdapter.get/libCurl .............................***Failed    0.17 sec
2021-03-16T16:46:47.4673939Z test 140
2021-03-16T16:46:47.4674842Z         Start 140: azure-core.Test/TransportAdapter.get204/libCurl
2021-03-16T16:46:47.4675361Z 
2021-03-16T16:46:47.4676585Z 140: Test command: /mnt/vss/_work/1/s/build/sdk/core/azure-core/test/ut/azure-core-test "--gtest_filter=Test/TransportAdapter.get204/libCurl" "--gtest_also_run_disabled_tests"
2021-03-16T16:46:47.4677610Z 140: Test timeout computed to be: 10000000
2021-03-16T16:46:47.4699517Z 140: Note: Google Test filter = Test/TransportAdapter.get204/libCurl
2021-03-16T16:46:47.4700214Z 140: [==========] Running 1 test from 1 test suite.
2021-03-16T16:46:47.4701149Z 140: [----------] Global test environment set-up.
2021-03-16T16:46:47.4701847Z 140: [----------] 1 test from Test/TransportAdapter
2021-03-16T16:46:47.4702356Z 140: [ RUN      ] Test/TransportAdapter.get204/libCurl
2021-03-16T16:46:47.6319730Z 140: unknown file: Failure
2021-03-16T16:46:47.6322604Z 140: C++ exception with description "Fail to get a new connection for: mt3.google.com. Couldn't resolve host name" thrown in the test body.
2021-03-16T16:46:47.6324665Z 140: [  FAILED  ] Test/TransportAdapter.get204/libCurl, where GetParam() = 48-byte object <D0-34 75-2D 9E-55 00-00 07-00 00-00 00-00 00-00 6C-69 62-43 75-72 6C-00 69-63 61-74 65-00 00-00 30-0A 75-2D 9E-55 00-00 20-0A 75-2D 9E-55 00-00> (162 ms)
2021-03-16T16:46:47.6328227Z 140: [----------] 1 test from Test/TransportAdapter (162 ms total)
2021-03-16T16:46:47.6329187Z 140: 
2021-03-16T16:46:47.6330248Z 140: [----------] Global test environment tear-down
2021-03-16T16:46:47.6331221Z 140: [==========] 1 test from 1 test suite ran. (162 ms total)
2021-03-16T16:46:47.6332086Z 140: [  PASSED  ] 0 tests.
2021-03-16T16:46:47.6332896Z 140: [  FAILED  ] 1 test, listed below:
2021-03-16T16:46:47.6334620Z 140: [  FAILED  ] Test/TransportAdapter.get204/libCurl, where GetParam() = 48-byte object <D0-34 75-2D 9E-55 00-00 07-00 00-00 00-00 00-00 6C-69 62-43 75-72 6C-00 69-63 61-74 65-00 00-00 30-0A 75-2D 9E-55 00-00 20-0A 75-2D 9E-55 00-00>
2021-03-16T16:46:47.6335858Z 140: 
2021-03-16T16:46:47.6336616Z 140:  1 FAILED TEST
2021-03-16T16:46:47.6338069Z 140/165 Test #140: azure-core.Test/TransportAdapter.get204/libCurl ..........................***Failed    0.17 sec
```

```text
The following tests FAILED:
	139 - azure-core.Test/TransportAdapter.get/libCurl (Failed)
	140 - azure-core.Test/TransportAdapter.get204/libCurl (Failed)
	141 - azure-core.Test/TransportAdapter.getLoop/libCurl (Failed)
	142 - azure-core.Test/TransportAdapter.head/libCurl (Failed)
	143 - azure-core.Test/TransportAdapter.put/libCurl (Failed)
	144 - azure-core.Test/TransportAdapter.deleteRequest/libCurl (Failed)
	145 - azure-core.Test/TransportAdapter.patch/libCurl (Failed)
	146 - azure-core.Test/TransportAdapter.getChunk/libCurl (Failed)
	147 - azure-core.Test/TransportAdapter.putErrorResponse/libCurl (Failed)
	148 - azure-core.Test/TransportAdapter.getWithStream/libCurl (Failed)
	149 - azure-core.Test/TransportAdapter.getLoopWithStream/libCurl (Failed)
	150 - azure-core.Test/TransportAdapter.headWithStream/libCurl (Failed)
	151 - azure-core.Test/TransportAdapter.putWithStream/libCurl (Failed)

```

**Comment by Jinming-Hu** [comment]
Don't merge breaking changes until storage release is done.

**Review by antkmsft — APPROVED** [review]

**Pull request merged**